flair/trainers/trainer.py

```diff
@@ -135,18 +147,22 @@ def train(self,
                     log_line(log)
                     break
 
-                if not test_mode:
-                    random.shuffle(train_data)
+                #if not test_mode:
+                #    random.shuffle(train_data)
```

Comment:
We shuffle the training data at each epoch so that the model does not overfit to a specific order of data points - this often improves model quality by quite a bit. Is there a way to shuffle the data at each epoch with the iterators?

Reply:
One way of doing this is to iterate the dataset and for each sentence or tag (or whatever entity we want to iterate) and we would also have char number as the start of the entity. Then we can random shuffle indexes which also point to a place in text file. Basically, implement random access. But the index itself might be OOM structure. Another option is to prefetch a batch of samples and random shuffle withing that batch.